presto-iceberg/src/main/java/com/facebook/presto/iceberg/rest/IcebergRestCatalogFactory.java

```diff
@@ -77,22 +126,99 @@ protected Map<String, String> getCatalogProperties(ConnectorSession session)
 
         catalogConfig.getSessionType().ifPresent(type -> {
             if (type == USER) {
-                properties.putAll(session.getIdentity().getExtraCredentials());
+                properties.put(CatalogProperties.USER, session.getUser());
+            }
+        });
 
+        return properties.build();
+    }
+
+    protected SessionContext convertSession(ConnectorSession session)
+    {
+        RestSessionBuilder sessionContextBuilder = builder(session);
+        catalogConfig.getSessionType().ifPresent(type -> {
+            if (type.equals(USER)) {
```

Comment:
to stick with the functional style of the `Optional` type, I would recommend instead to use `.filter(type -> type.equals(USER))`, then you can remove the conditional here in the `ifPresent` block

Reply:
Ah thank you! I completely forgot about `filter` (clearly)